gitpod-io/gitpod
# support code cli

#### What it does

- enables IDE images to contribute env variables via special scheme:
```dockerfile
# to prepend a value to KEY variable value
ENV GITPOD_ENV_PREPEND_KEY "value" 
# to append a value to KEY variable value
ENV GITPOD_ENV_APPEND_KEY "value" 
# to set a value for KEY variable
ENV GITPOD_ENV_SET_KEY "value" 
```
- implements code CLI: https://github.com/gitpod-io/vscode/compare/20ded0d...785b788
- configure Gitpod Code image using env variables.

#### How to test

- With disabled feature preview start a workspace and check that `gp`, `code` and `open` are available as well as `EDITOR`, `VISUAL` and `GIT_EDITOR` env variables are configured to use them.
  - Check both in task terminals started by the supervisor and regular terminals started by Theia. It should be true for tests below as well.
- Enable feature preview and start a workspace with as well. Do checks again.
- Now switch to Code as a default IDE and start a new workspace. Check taht `code` command does not delegate to `gp` anymore. Check that `code`, `open` and `gitpod-code` do the same. Check that env variables are configured properly.
  - Don't forget that different options of code command actually work.
